### PR TITLE
Check that dates for a report have the same month

### PR DESCRIPTION
**Pull request by vbelinschi**
## References

- #0000

### CODE REVIEW

**Review by dzhavat — APPROVED** [review]

File: libs/dh/globalization/assets-localization/src/assets/i18n/en.json [hunk 1]
```diff
@@ -1368,6 +1368,7 @@
         "continue": "Next",
         "confirm": "{{ wholesale.settlementReports.requestSettlementReport }}",
         "startDateIsBeforeMinDate": "Start date cannot be older than 3,6 years + the current month",
+        "startDateAndEndDateHaveDifferentMonth": "Start date and end date should have the same month",
```
Comment (michaelmajgaard) [inline libs/dh/globalization/assets-localization/src/assets/i18n/en.json:1371]: Suggestion: "Start and end date must be within the same month."

File: libs/dh/globalization/assets-localization/src/assets/i18n/da.json [hunk 2]
```diff
@@ -1368,6 +1368,7 @@
         "confirm": "{{ wholesale.settlementReports.requestSettlementReport }}",
         "continue": "Næste",
         "startDateIsBeforeMinDate": "Startdatoen kan ikke være ældre end 3,6 år + løbende måned",
+        "startDateAndEndDateHaveDifferentMonth": "Startdatoen og slutdatoen skal have det samme måned",
```
Comment (michaelmajgaard) [inline libs/dh/globalization/assets-localization/src/assets/i18n/da.json:1371]: Suggestion: "Start- og slutdato skal være inde for samme måned."

File: apps/dh/api-dh/source/DataHub.WebApi/GraphQL/Mutation/SettlementReportMutation.cs [hunk 3]
```diff
@@ -63,4 +69,12 @@ public async Task<bool> CancelSettlementReportAsync(
 
         return true;
     }
+
+    private bool IsPeriodAcrossMonths(Interval interval)
```
Comment (michaelmajgaard) [inline apps/dh/api-dh/source/DataHub.WebApi/GraphQL/Mutation/SettlementReportMutation.cs:73]: Also might be a good idea to write a unit test for this one, with a few different cases. Method should also be static.

File: apps/dh/api-dh/source/DataHub.WebApi.Tests/Unit/Clients/Wholesale/SettlementReports/SettlementReportsClientTests.cs [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+    public SettlementReportsClientTests()
+    {
+        _httpMessageHandlerMock = new Mock<HttpMessageHandler>();
+        _httpClient = new HttpClient(_httpMessageHandlerMock.Object);
+        _httpClient.BaseAddress = new Uri("http://localhost/");
+        _client = new SettlementReportsClient("http://localhost", _httpClient, _httpClient, _httpClient);
+    }
+
+    [Theory]
+    [InlineData("2024-11-01T23:00:00.000Z", "2024-12-31T22:59:59.999Z")]
```
Comment (michaelmajgaard) [inline apps/dh/api-dh/source/DataHub.WebApi.Tests/Unit/Clients/Wholesale/SettlementReports/SettlementReportsClientTests.cs:47]: Doesn't the frontend send `2024-12-31T23:00:00Z` as the to date, if not, the `AddMilliseconds(-1)` does not make sense?

**Pull request merged**